info/v1/metric.go

```diff
@@ -32,38 +32,21 @@ const (
 	MetricDelta = "delta"
 )
 
-// An exported metric.
-type Metric struct {
+// Spec for custom metric.
+type MetricSpec struct {
 	// The name of the metric.
 	Name string `json:"name"`
 
 	// Type of the metric.
 	Type MetricType `json:"type"`
-
-	// Metadata associated with this metric.
-	Labels map[string]string
```

Comment:
I see. Wouldn't we have naming conflicts if we have two stats series that are both called disk_usage and disambiguation depends on the label. I would lean towards pushing that kind of info in the name, in a structured way.

I think this will be relevant when we implement a prometheus reader. Let's revisit it then.